File: Common_Character_Count.py
# Arcade > Intro > 10
# Given two strings, find the number of common characters between them
# Ex input: s1 = "aabcc" and s2 = "adcaa" => 3

def commonCharacterCount(s1, s2):
    from collections import Counter
    o = Counter(s1)&Counter(s2) #object containing common letters and their counts, ex Counter({'a': 2, 'c': 1})
    e = list(a.elements()) #list of all elements, ex ['a', 'a', 'c']
    return len(e)

File: CodeSignalJS/Common_Character_Count.js
// Arcade > Intro > 10
// Given two strings, find the number of common characters between them
// Ex input: s1 = "aabcc" and s2 = "adcaa" => 3

function commonCharacterCount(s1, s2) {
    for (var i=0; i<s1.length; i++) { //iterates through s1
        s2= s2.replace(s1[i],"*") //replaces up to one instance of s1[i] with "*"
    }
    return s2.replace(/[^*]/g,"").length //erases all non "*" characters (all differences)
}